janino/src/main/java/org/codehaus/janino/UnitCompiler.java

```diff
@@ -9356,50 +9356,52 @@
 
         if (maximallySpecificIInvocables.size() == 1) return (IInvocable) maximallySpecificIInvocables.get(0);
 
-        ONE_NON_ABSTRACT_INVOCABLE:
+        ONE_CONCRETE_INVOCABLE:
         if (maximallySpecificIInvocables.size() > 1 && iInvocables[0] instanceof IClass.IMethod) {
 
             // Check if all methods have the same signature (i.e. the types of all their parameters are identical) and
-            // exactly one of the methods is non-abstract (JLS7 15.12.2.2.BL2.B1).
-            IClass.IMethod theNonAbstractMethod = null;
+            // exactly one of the methods is concrete (JLS8 15.12.2.5.B8.B2).
+            IClass.IMethod theConcreteMethod = null;
             {
                 Iterator<IClass.IInvocable> it                          = maximallySpecificIInvocables.iterator();
                 IClass.IMethod              m                           = (IClass.IMethod) it.next();
                 final IClass[]              parameterTypesOfFirstMethod = m.getParameterTypes();
                 for (;;) {
-                    if (!m.isAbstract()) {
-                        if (theNonAbstractMethod == null) {
-                            theNonAbstractMethod = m;
+                    if (!m.isAbstract() && !m.getDeclaringIClass().isInterface()) {

```

Comment:
@aunkrig pardon me for pinging you, but I wanted to see what you thought of this change. Really, it could be boiled down to this single line change (the rest is almost entirely doc/name updates). The upside is that this change, at least, allows Spark to work with janino on 2.12, and possibly other Scala 2.12 apps. I suppose it needs to be limited to Java 8 contexts only, and I didn't know if there was an easy way to do that inside janino other than looking at the Java version. But in that case I think there's no downside in Java 8. WDYT? thank you!